fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerializerGenerator.java

```diff
@@ -296,12 +293,12 @@ private void processUnion(final Schema unionSchema, JExpression unionExpr, JBloc
 
     for (Schema schemaOption : unionSchema.getTypes()) {
       if (Schema.Type.NULL.equals(schemaOption.getType())) {
-        /**
+        /*
          * We always handle the null branch of the union first, otherwise, it leads to a bug in the
          * case where there is an optional field where the null is the second branch of the union.
          */
         JExpression condition = unionExpr.eq(JExpr._null());
-        ifBlock = ifBlock != null ? ifBlock._elseif(condition) : body._if(condition);
+        ifBlock = body._if(condition);
```

Comment:
What is the rationale for this change?

Reply:
Sure. Let's consider two key facts:
- For the first time we enter this block `ifBlock` is `null`
- The block can be entered at most once due to `break` at the end.

It means condition `ifBlock != null` is always `false`.
"Always" in this context means "at most once".

Reply:
I see. That's right... I missed that detail when I refactored this code in https://github.com/linkedin/avro-util/commit/608bce84afc4af19fc6d9d70b0b8be02afa477df